hibernate-core/src/main/java/org/hibernate/type/TypeHelper.java

```diff
@@ -90,6 +91,33 @@ else if ( target[i] == LazyPropertyInitializer.UNFETCHED_PROPERTY ) {
 		return copied;
 	}
 
+	/**
+	 * Apply the {@link Type#replace} operation across a series of values.
+	 *
+	 * @param persister The EntityPersister
+	 * @param entity The source of the state
+	 * @param session The originating session
+	 * @param owner The entity "owning" the values
+	 * @param copyCache A map representing a cache of already replaced state
+	 *
+	 */
+	public static void replace(
+			final EntityPersister persister,
+			final Object entity,
+			final SharedSessionContractImplementor session,
+			final Object owner,
+			final Map<Object, Object> copyCache) {
+		final Object[] values = persister.getValues( entity );
+		final Type[] types = persister.getPropertyTypes();
+		for ( int i = 0; i < types.length; i++ ) {
+			if ( values[i] != LazyPropertyInitializer.UNFETCHED_PROPERTY
+					&& values[i] != PropertyAccessStrategyBackRefImpl.UNKNOWN
+					&& !( types[i] instanceof AbstractStandardBasicType ) ) {
+				persister.setValue( entity, i, types[i].replace( values[i], values[i], session, owner, copyCache ) );
+			}
```

Comment:
I guess the check `!( types[i] instanceof AbstractStandardBasicType )` is to determine if replace will actually do something? I don't think this is correct for mutable types like `java.util.Date`, so IMO the more correct approach is this:

```suggestion
			final Object oldValue = values[i];
			final Object newValue;
			if ( oldValue != LazyPropertyInitializer.UNFETCHED_PROPERTY
					&& oldValue != PropertyAccessStrategyBackRefImpl.UNKNOWN
					&& ( newValue = types[i].replace( values[i], values[i], session, owner, copyCache ) ) != oldValue ) {
				persister.setValue( entity, i, newValue );
			}
```